TGEngine/io/Resource.cpp

```diff
@@ -10,9 +11,140 @@ namespace tge::io {
+			ActorProperties* currentProperty = actorProperties.data() + currentId;
+			ActorDescriptor* currentDescription = actorDescriptor.data() + currentId;
+
+			fread(&currentProperty->localTransform, sizeof(float), 16, file);
+			fread(&currentProperty->material, sizeof(uint8_t), 1, file);
+			fread(&currentProperty->layer, sizeof(uint8_t), 1, file);
+
+			uint32_t vertexCount = 0;
+			fread(&currentDescription->indexDrawCount, sizeof(uint32_t), 1, file);
+			fread(&vertexCount, sizeof(uint32_t), 1, file);
+
+			currentDescription->indexOffset = lastIndexCount;
+			currentDescription->vertexOffset = lastVertexCount;
+			lastIndexCount += currentDescription->indexDrawCount;
+			lastVertexCount += vertexCount;
+
+			indexBufferCopy[currentId].dstOffset = lastIndexOffset;
+			indexBufferCopy[currentId].srcOffset = 0;
+			vertexBufferCopy[currentId].dstOffset = lastVertexOffset;
+			vertexBufferCopy[currentId].srcOffset = 0;
+
+			// Creating staging buffer for indices and vertices
+			BufferInputInfo bufferInputInfos[2];
+			bufferInputInfos[0].flags = VK_SHADER_STAGE_ALL_GRAPHICS;
+			bufferInputInfos[0].size = sizeof(uint32_t) * currentDescription->indexDrawCount;
+			bufferInputInfos[0].memoryIndex = vlibDeviceHostVisibleCoherentIndex;
+			bufferInputInfos[0].bufferUsageFlag = VK_BUFFER_USAGE_STORAGE_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_SRC_BIT;
+
+			bufferInputInfos[1].flags = VK_SHADER_STAGE_ALL_GRAPHICS;
+			bufferInputInfos[1].size = blocklength;
+			bufferInputInfos[1].memoryIndex = vlibDeviceHostVisibleCoherentIndex;
+			bufferInputInfos[1].bufferUsageFlag = VK_BUFFER_USAGE_STORAGE_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_SRC_BIT;
+
+			lastIndexOffset += (indexBufferCopy[currentId].size = bufferInputInfos[0].size);
+			lastVertexOffset += (vertexBufferCopy[currentId].size = blocklength);
+
+			createBuffers(bufferInputInfos, 2, &buffer[currentId * 2]);
+
+			void* indexMemory;
+			CHECKFAIL(vkMapMemory(device, buffer[currentId * 2].memory, 0, VK_WHOLE_SIZE, 0, &indexMemory));
+
+			// Reads the indices from the file
+			fread(indexMemory, sizeof(uint32_t), currentDescription->indexDrawCount, file);
+
+			vkUnmapMemory(device, buffer[currentId * 2].memory);
+
+			void* vertexMemory;
+			CHECKFAIL(vkMapMemory(device, buffer[currentId * 2 + 1].memory, 0, VK_WHOLE_SIZE, 0, &vertexMemory));
+
+			// Reads the vertices
+			fread(vertexMemory, sizeof(uint8_t), blocklength, file);
+
+			vkUnmapMemory(device, buffer[currentId * 2 + 1].memory);
+
+			// Read the next block size
+			fread(&blocklength, sizeof(uint32_t), 1, file);
+		}
+
+#ifdef DEBUG
+		if (lastIndexOffset == 0) {
+			OUT_LV_DEBUG("It seams like you are missing indices!")
+		}
+		if (lastVertexOffset == 0) {
+			OUT_LV_DEBUG("It seams like you are missing vertices")
+		}
+		if (lastIndexCount == 0) {
+			OUT_LV_DEBUG("It seams like you are missing indices!")
+		}
+		if (lastVertexCount == 0) {
+			OUT_LV_DEBUG("It seams like you are missing vertices")
+		}
```

Comment:
```suggestion
		if (lastIndexOffset == 0 || lastIndexCount == 0) {
			OUT_LV_DEBUG("It seems like you are missing indices!")
		}
		if (lastVertexOffset == 0 || lastVertexCount == 0) {
			OUT_LV_DEBUG("It seems like you are missing vertices!")
		}
```

mind the "seams" typo before you're manually editing instead of copypasting